### PR TITLE
CVS-46631 statefult rest client

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: example_client/rest_stateful_client.py [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+        print(
+            "ERROR: Number of input ark files {} must be equal to the number of input names {}".format(
+                len(input_files),
+                len(input_names)))
+        exit(1)
+
+    if len(reference_files) != len(output_names):
+        print(
+            "ERROR: Number of output ark files {} must be equal to the number of output names {}".format(
+                len(input_files),
```
Comment (mzegla) [inline example_client/rest_stateful_client.py:190]: reference_files*

File: example_client/rest_stateful_client.py [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+                len(input_names)))
```
Comment (mzegla) [inline example_client/rest_stateful_client.py:191]: output_names*

File: example_client/rest_stateful_client.py [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+                print("The server response is not json format: {}",format(result.text))
+                exit(1)
+            if "error" in result_dict:
+                print('Server returned error: {}'.format(result_dict))
+                exit(1)
+
+            if not validate_output(result_dict, output_names):
+                print(
+                    "ERROR: Model result validation error. Adding end sequence inference request for the model and exiting.")
+                # request.inputs['sequence_control_input'].CopyFrom(
```
Comment (michalkulakowski) [inline example_client/rest_stateful_client.py:387]: Why is this code commented out?

Comment (rasapala) [reply example_client/rest_stateful_client.py]: Removed, was used in grpc.

Comment (rasapala) [reply example_client/rest_stateful_client.py]: Removed, was used in grpc.

**Review by michalkulakowski — APPROVED** [review]

File: example_client/rest_stateful_client.py [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+            score_index += 1
+            # END utterance loop
+
+        seq_avg_rms_error_sum = mean_avg_rms_error_sum / (sequence_size)
+        print(
+            "\tSequence {} average rms error: {:.10f}\n".format(
+                sequence_id,
+                seq_avg_rms_error_sum))
+        sequence_id += 1
```
Comment (mzegla) [inline example_client/rest_stateful_client.py:436]: Can we please get rid of this line? I think as we let user define sequence id in CLI argument we should stick to it the entire time.

File: example_client/rest_stateful_client.py [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+                            np.average(duration), 2), round(
+                            1000 * sequence_size / np.average(duration), 2)))
+                # END output names loop
+
+            score_index += 1
+            # END utterance loop
+
+        seq_avg_rms_error_sum = mean_avg_rms_error_sum / (sequence_size)
+        print(
+            "\tSequence {} average rms error: {:.10f}\n".format(
```
Comment (mzegla) [inline example_client/rest_stateful_client.py:433]: Can we also print sequence name here like we do in grpc_stateful_client.py?

**Review by mzegla — APPROVED** [review]

**Pull request merged**